pkg/spoke/submarineragent/config_controller.go

```diff
@@ -181,6 +189,46 @@ func (c *submarinerConfigController) sync(ctx context.Context, syncCtx factory.S
 		return c.updateGatewayStatus(syncCtx.Recorder(), config)
 	}
 
+	if config.Status.ManagedClusterInfo.Platform == "GCP" {
+		if !helpers.IsSubmarinerEnvPrepared(c.configClient, config.Namespace, config.Name) {
```

Comment:
This check doesn't seem necessary.

Reply:
This was marked as resolved but wasn't. Please remove this call and the function.